internal/services/zero_trust_access_mtls_certificate/resource_test.go

```diff
@@ -91,24 +111,43 @@ func testSweepCloudflareAccessMutualTLSCertificate(r string) error {
 			continue
 		}
 
-		if len(cert.AssociatedHostnames) > 0 {
-			// to delete we need to update first with empty hostnames
-			_, err = client.UpdateAccessMutualTLSCertificate(context.Background(), cloudflare.ZoneIdentifier(zoneID), cloudflare.UpdateAccessMutualTLSCertificateParams{
-				ID:                  cert.ID,
-				AssociatedHostnames: []string{},
-			})
+		// to delete we need to update first with empty hostnames
+		_, err = client.UpdateAccessMutualTLSCertificate(context.Background(), cloudflare.ZoneIdentifier(zoneID), cloudflare.UpdateAccessMutualTLSCertificateParams{
+			ID:                  cert.ID,
+			AssociatedHostnames: []string{},
+		})
 
-			if err != nil {
-				tflog.Error(ctx, fmt.Sprintf("Failed to update Cloudflare Access Mutual TLS certificate (%s) in zone ID: %s", cert.ID, zoneID))
-				return err
+		if err != nil {
+			tflog.Error(ctx, fmt.Sprintf("Failed to update Cloudflare Access Mutual TLS certificate (%s) in zone ID: %s", cert.ID, zoneID))
+		}
+
+		// Wait for update to propagate with retry logic
+		maxRetries := 5
+		backoff := time.Second * 2
+		for range maxRetries {
+			time.Sleep(backoff)
+			updatedCert, checkErr := client.GetAccessMutualTLSCertificate(context.Background(), cloudflare.ZoneIdentifier(zoneID), cert.ID)
+			if checkErr == nil && len(updatedCert.AssociatedHostnames) == 0 {
+				break
 			}
+			backoff *= 2
 		}
 
-		err := client.DeleteAccessMutualTLSCertificate(context.Background(), cloudflare.ZoneIdentifier(zoneID), cert.ID)
+		// Retry deletion with exponential backoff to handle race conditions
+		var lastErr error
+		for range maxRetries {
+			err = client.DeleteAccessMutualTLSCertificate(context.Background(), cloudflare.ZoneIdentifier(zoneID), cert.ID)
+			if err == nil {
+				lastErr = nil
+				break
+			}
+			lastErr = err
+			time.Sleep(backoff)
+			backoff *= 2
+		}
 
-		if err != nil {
-			tflog.Error(ctx, fmt.Sprintf("Failed to delete Cloudflare Access Mutual TLS certificate (%s) in zone ID: %s", cert.ID, zoneID))
-			return err
+		if lastErr != nil {
+			tflog.Error(ctx, fmt.Sprintf("Failed to delete Cloudflare Access Mutual TLS certificate (%s) in zone ID: %s after retries: %s", cert.ID, zoneID, lastErr))
```

Comment:
if this fails, wont the hostnames will  emptied but the state will still have them, leading to issues?